### PR TITLE
Fix Ivy + Gradle Module Metadata skipping on Gradle 6+

### LINKED ISSUE #284: build-info-extractor-gradle breaks with Gradle 6.x when Gradle Module Metadata publication is disabled
Hi folks,

Gradle Module Metadata is a format used to serialize the Gradle component model that was introduced as default in Gradle 6.x. However, still possible to disable this for multiple reasons: https://docs.gradle.org/current/userguide/publishing_gradle_module_metadata.html#sub:disabling-gmm-publication

For example

```
tasks.withType(GenerateModuleMetadata) {
        enabled = false
    }
```

When this is disabled, 4.11.0 version of the plugin fails to publish with exceptions such as:

```Caused by: java.lang.RuntimeException: java.util.concurrent.ExecutionException: org.gradle.api.GradleException: File '/agent/workspace/NEBULA-nebula/subprojects/core/build/nebulatest/netflix.nebula.publishing.NebulaReleasePublishLauncherSpec/publish-a-jar/build/publications/nebulaIvy/module.json' does not exist, and need to be published from publication nebulaIvy```

While debugging found that ivyPublicationInternal.getPublishableArtifacts() has the module.json file on it

```
0 = {ArchiveTaskBasedIvyArtifact_Decorated@16692} "ArchiveTaskBasedIvyArtifact_Decorated testnebulareleasepublish:sources:jar:sources"
1 = {PublishArtifactBasedIvyArtifact_Decorated@17562} "PublishArtifactBasedIvyArtifact_Decorated testnebulareleasepublish:jar:jar:null"
2 = {SingleOutputTaskIvyArtifact@17563} "SingleOutputTaskIvyArtifact ivy:ivy:xml:null"
3 = {SingleOutputTaskIvyArtifact@17564} "SingleOutputTaskIvyArtifact testnebulareleasepublish:json:module:null"
```

This was introduced with https://github.com/jfrog/build-info/pull/265/files. 

Ideally, the plugin shouldn't try to publish files that are not being generated

Opening this ticket for visibility and already gave a heads up to @ljacomet

### PR DESCRIPTION
**Pull request by ljacomet**
This fixes #284 and also addresses the Maven publication issue raised in #271

### CODE REVIEW

File: build-info-extractor-gradle/src/main/groovy/org/jfrog/gradle/plugin/artifactory/task/helper/TaskHelperPublications.java [hunk 1]
```diff
@@ -261,10 +261,15 @@ public boolean hasModules() {
             } catch (NoSuchMethodError error) {
                 // Compatibility with older versions of Gradle:
                 artifacts = mavenNormalizedPublication.getAllArtifacts();
+                legacy = true;
             }
 
             // Third adding all additional artifacts - includes Gradle Module Metadata when produced
             for (MavenArtifact artifact : artifacts) {
+                if (legacy && artifact.getFile().equals(pomFile)) {
+                    // Need to skip the POM file for Gralde < 5.0
```
Comment (OdysseusLives) [inline build-info-extractor-gradle/src/main/groovy/org/jfrog/gradle/plugin/artifactory/task/helper/TaskHelperPublications.java]: Spelling change needed :)

Comment (ljacomet) [reply build-info-extractor-gradle/src/main/groovy/org/jfrog/gradle/plugin/artifactory/task/helper/TaskHelperPublications.java]: Fixed, thanks!

**Comment by ljacomet** [comment]
@yahavi I saw in #287 that you plan a release. Is there a chance to get this merged as well? Thanks!

**Comment by yahavi** [comment]
@ljacomet Yes. I'll review it as soon as possible.

**Comment by yahavi** [comment]
@ljacomet I tried [this project](https://github.com/jfrog/project-examples/tree/master/gradle-examples/gradle-example-publish) with Gradle 4.10.3 and the poms still got published.
The following configuration used, after installing the build-info project:

```
repositories {
   mavenLocal()
   jcenter()
}
dependencies {
   classpath(group: 'org.jfrog.buildinfo', name: 'build-info-extractor-gradle', version: '4.9.x-SNAPSHOT')
}
```
[publishPom](https://github.com/jfrog/project-examples/blob/master/gradle-examples/gradle-example-publish/build.gradle#L128) changed to false.

With the same configuration, the poms didn't publish with Gradle 5.6.4 and Gradle 6.0.1.
Can you please check?
Thanks!

**Pull request merged**